src/Aspire.Hosting.SqlServer/SqlServerBuilderExtensions.cs

```diff
@@ -45,6 +47,42 @@ public static IResourceBuilder<SqlServerServerResource> AddSqlServer(this IDistr
             }
         });
 
+        builder.Eventing.Subscribe<ResourceReadyEvent>(sqlServer, async (@event, ct) =>
+        {
+            if (connectionString is null)
+            {
+                throw new DistributedApplicationException($"ResourceReadyEvent was published for the '{sqlServer.Name}' resource but the connection string was null.");
+            }
+
+            using var sqlConnection = new SqlConnection(connectionString);
+            await sqlConnection.OpenAsync(ct).ConfigureAwait(false);
+
+            foreach (var sqlDatabase in sqlServer.DatabaseResources)
+            {
+                var quotedDatabaseIdentifier = new SqlCommandBuilder().QuoteIdentifier(sqlDatabase.DatabaseName);
+
+                try
+                {
+                    if (sqlConnection.State != System.Data.ConnectionState.Open)
+                    {
+                        throw new InvalidOperationException($"Could not open connection to '{sqlServer.Name}'");
+                    }
+
+                    var scriptAnnotation = sqlDatabase.Annotations.OfType<ScriptAnnotation>().LastOrDefault();
+
+                    using var command = sqlConnection.CreateCommand();
+                    command.CommandText = scriptAnnotation?.Script ??
+                        $"IF ( NOT EXISTS ( SELECT 1 FROM sys.databases WHERE name = @DatabaseName ) ) CREATE DATABASE {quotedDatabaseIdentifier};";
```

Comment:
Why the difference between the name in the WHERE and the name used on the Create Database? Shouldn't they both use the same SqlParameter?

Reply:
Parameters can't be used for object names (such as database names, table names, or column names). 
It's used in the query part because in that case it's a string value.